src/state/statePoller.ts

```diff
@@ -0,0 +1,61 @@
+import { Disposable, Event, EventEmitter } from "vscode";
+import { State } from "../binary/state";
+
+import { BINARY_STATE_POLLING_INTERVAL_MILLISECONDS } from "../globals/consts";
+import { getState, tabNineProcess } from "../binary/requests/requests";
+
+export type StateEmitterProps = {
+  currentState: State | null | undefined;
+  previousState: State | null | undefined;
+};
+
+export class StatePoller implements Disposable {
```

Comment:
should be a singleton to avoid all the code locations doing their own polling. The polling interval should be the LCD of all of them